  @Test
  public void testName() throws Exception {
    CompletableFuture<MessageActionItem> future = new CompletableFuture<>();
    when(transmitter.sendShowMessageRequest(any())).thenReturn(future);

    CheLanguageClient client = new CheLanguageClient(eventService, transmitter, "id");
    server = launcher.launch("file:///tmp", client);
    InitializeParams params = new InitializeParams();
    params.setRootUri("file:///tmp");
    params.setCapabilities(new ClientCapabilities());
    server.initialize(params);
    ArgumentCaptor<ShowMessageRequestParams> captor =
        ArgumentCaptor.forClass(ShowMessageRequestParams.class);
    verify(transmitter, timeout(1500)).sendShowMessageRequest(captor.capture());

    ShowMessageRequestParams value = captor.getValue();
    assertNotNull(value);
    assertEquals(value.getType(), MessageType.Error);
    assertEquals(value.getMessage(), "Error Message!!!!");
  }

